### PR TITLE
Added xsd:dateTimeStamp, initial patch provided by Toni Cebrián

### PR DESCRIPTION
**Pull request by barthanssens**
GitHub issue resolved: #1729  

Briefly describe the changes proposed in this PR:
* Added constant to XMLSchema vocabulary
* Added checks in XmlDateUtil
* Added tests

### CODE REVIEW

**Comment by hmottestad** [comment]
@tonicebrian

**Comment by hmottestad** [comment]
One quick question. Does this also support sorting? ie. by using the default RDF4J comparator.

**Comment by barthanssens** [comment]
Good catch, I'll write a compareDateTimeStamp method and add it to compare

**Comment by barthanssens** [comment]
Hmz, JDK 8 (and 11) only supports XML Schema 1.0, not 1.1 (https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/javax/xml/datatype/DatatypeConstants.html)

Doesn't stop this to be useful, but there might be cases where functionality is limited.

**Comment by barthanssens** [comment]
My main (minor) concern is that one has to use 
`createLiteral(String datetime, XMLSchema.DATETIMESTAMP)`

Because `javax.xml.datatype.XMLGregorianCalendar` and `javax.xml.datatype.DatatypeConstants` don't support XML Schema 1.1 datatypes

**Comment by barthanssens** [comment]
@tonicebrian: I think it should be fairly complete now, so basically ready to merge unless I forgot something

File: core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java [hunk 1]
```diff
@@ -1809,8 +1823,9 @@ private static String trimPlusSign(String s) {
 	}
 
 	/**
-	 * Maps a datatype QName from the javax.xml.namespace package to an XML Schema URI for the corresponding datatype.
-	 * This method recognizes the XML Schema qname mentioned in {@link DatatypeConstants}.
+	 * Maps a datatype QName from the javax.xml.namespace package to an XML Schema 1.0 URI for the corresponding
+	 * datatype. This method recognizes the XML Schema qname mentioned in {@link DatatypeConstants}. Note that Java 8 /
+	 * 11 do not have constants for XML Schema 1.0 datatypes like xsd:dateTimeStamp.
```
Comment (jeenbroekstra) [inline core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java]: Do you mean XML Schema 1.1 here?

Comment (barthanssens) [reply core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java]: ah indeed, thanks

**Comment by tonicebrian** [comment]
@barthanssens LGTM, Thanks!!

**Pull request merged**